ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -121,33 +115,25 @@ private[ember] object Encoder {
               .append(CRLF)
           }
 
-        req.headers
-          .get[`Content-Length`]
-          .foreach { _ =>
-            appliedContentLength = true
-          }
-
         // Apply each header followed by a CRLF
         req.headers.foreach { h =>
           if (h.isNameValid) {
-            stringBuilder
-              .append(h.name)
-              .append(": ")
+            appliedContentLength = appliedContentLength || h.name == `Content-Length`.name
+
+            stringBuilder.append(h.name).append(": ")
             appendSanitized(stringBuilder, h.value)
             stringBuilder.append(CRLF)
-            ()
           }
         }
 
-        if (
-          !appliedContentLength && req.body == EmptyBody && !NoPayloadMethods.contains(req.method)
-        ) {
+        def isEmptyBody = req.body eq EmptyBody
+        def isNoPayloadMethod = NoPayloadMethods.contains(req.method)
+        if (!appliedContentLength && isEmptyBody && !isNoPayloadMethod) {
```

Comment:
Continuing bike-shedding, speaking in terms of saving a few ns, shouldn't we place `!isNoPayloadMethod` ahead of `isEmptyBody`? I mean, `!isNoPayloadMethod` might be `false` a bit frequently then `isEmptyBody`.

Reply:
That might be, but I think reference equality `eq` is faster than set membership `.contains`.